hedged.go

```diff
@@ -42,21 +56,31 @@ func New(cfg Config) (*Client, error) {
 		return nil, err
 	}
 
+	// TODO(cristaloleg): this should be removed after internals cleanup.
```

Comment:
How can it return not a `*hedgedTransport`? :thinking: maybe the cast could be added to the `rt` variable itself?